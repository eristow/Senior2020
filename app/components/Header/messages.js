/*
 * Header Messages
 *
 * This contains all the text for the Header component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.Header';

export default defineMessages({
  home: {
    id: `${scope}.home`,
    defaultMessage: 'Home',
  },
  drumMachine: {
    id: `${scope}.drumMachine`,
    defaultMessage: 'Drum Machine',
  },
  timeline: {
    id: `${scope}.timeline`,
    defaultMessage: 'Timeline',
  },
  piano: {
    id: `${scope}.piano`,
    defaultMessage: 'Piano',
  },
  drums: {
    id: `${scope}.drums`,
    defaultMessage: 'Drums',
  },
});
