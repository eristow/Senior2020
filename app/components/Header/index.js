/**
 *
 * Header
 *
 */

import React from 'react';
import { FormattedMessage } from 'react-intl';
import NavBar from './NavBar';
import HeaderLink from './HeaderLink';
import messages from './messages';

function Header() {
  return (
    <NavBar>
      <HeaderLink to="/">
        <FormattedMessage {...messages.home} />
      </HeaderLink>
      <HeaderLink to="/timeline">
        <FormattedMessage {...messages.timeline} />
      </HeaderLink>
      <HeaderLink to="/piano">
        <FormattedMessage {...messages.piano} />
      </HeaderLink>
      <HeaderLink to="/drums">
        <FormattedMessage {...messages.drums} />
      </HeaderLink>
    </NavBar>
  );
}

export default Header;
