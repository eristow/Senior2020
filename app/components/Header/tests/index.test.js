/**
 *
 * Tests for Header
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { Provider } from 'react-redux';
import { IntlProvider } from 'react-intl';
import { ConnectedRouter } from 'connected-react-router/immutable';
import { createMemoryHistory } from 'history';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import configureStore from '../../../configureStore';
import Header from '../index';
import { DEFAULT_LOCALE } from '../../../i18n';

describe('<Header />', () => {
  let history;
  let store;

  beforeAll(() => {
    history = createMemoryHistory();
    store = configureStore({}, history);
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <ConnectedRouter history={history}>
            <Header />
          </ConnectedRouter>
        </IntlProvider>
      </Provider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <ConnectedRouter history={history}>
            <Header />
          </ConnectedRouter>
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
