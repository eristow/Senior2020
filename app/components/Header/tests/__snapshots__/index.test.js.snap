// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Should render and match the snapshot 1`] = `
<div
  class="NavBar-sc-1a70pn8-0 fBbJXM"
>
  <a
    class="HeaderLink-sc-5k2h0m-0 fcQMtx"
    href="/"
  >
    <span>
      Home
    </span>
  </a>
  <a
    class="HeaderLink-sc-5k2h0m-0 fcQMtx"
    href="/timeline"
  >
    <span>
      Timeline
    </span>
  </a>
  <a
    class="HeaderLink-sc-5k2h0m-0 fcQMtx"
    href="/piano"
  >
    <span>
      Piano
    </span>
  </a>
  <a
    class="HeaderLink-sc-5k2h0m-0 fcQMtx"
    href="/drums"
  >
    <span>
      Drums
    </span>
  </a>
</div>
`;
